---
title: "BrdU RNA-seq"
author: "Andrew Ding-Su"
date: "2024-12-20"
output:
   pdf_document:
     extra_dependencies: ["float"]
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
knitr::opts_chunk$set(fig.pos = "H", out.extra = "")

#library calls
library("tidyverse")
library("variancePartition")
library("edgeR")
library("BiocParallel")
library("readxl")
library("gridExtra")

source("./scripts/1 normalize.R")
source("./scripts/2 analysis.R")
source("./scripts/3 graphing.R")

```

```{r libraries, include = FALSE}
#Load data
count_data <- read.csv("./data/countMatrix.csv")
animals <- read_xlsx("./data/library_info.xlsx") 
brdu_values <- read.csv("./data/brdu_values.csv")

```
```{r cleaning, include = FALSE}
animals <- animals %>%
  mutate(sample = `Sample no.`, 
         animal = factor(Animal), 
         timepoint = factor(Timepoint, levels = c("0","21dpi","nec")),
         cd8_depleted = factor(cd8_depleted, levels = c("N","Y")),
         infected = factor(infected, levels = c("N","Y")),
         aids = factor(aids, levels = c("N","Y")),
         sive = factor(sive, levels = c("N","Y")),
         plus_minus = factor(`plus/minus`, levels = c("minus","plus")),
         library_date = factor(library_date),
         sort_date = factor(sort_date),
         RNA_date = factor(RNA_date),
         .keep = "none") %>%
  
  #next steps are to get the sample names to match
  mutate(sample = ifelse(sample <= 9, paste0("0",sample), sample)) %>%
  mutate(sample = paste0("sample_",sample))
animals <- as.data.frame(animals)
rownames(animals) <- animals$sample

#fixing data from brdu_values
brdu_values <- brdu_values %>%
  mutate(sample = `Sample.no.`) %>%
  mutate(sample = ifelse(sample <= 9, paste0("0",sample), sample)) %>%
  mutate(sample = paste0("sample_",sample))
rownames(brdu_values) <- brdu_values$sample

#Create full metadata sheet by merging animals with brdu_values
animals_merged <- merge(animals, brdu_values, by=0, all=TRUE) %>%
  mutate(sample = sample.x,
         animal,
         timepoint,
         cd8_depleted,
         infected,
         aids,
         sive,
         plus_minus,
         library_date,
         sort_date,
         RNA_date,
         per_brdu,
         .keep = "none")
rownames(animals_merged) <- animals_merged$sample

#generate dgelist object for analyses
dge <- normalize_reads(count_data)
```

# Quality control - batch correction
\quad Technical, non-biological variation due to RNA extraction or sorting introduces batch effects. Batch effects obscure biological variation. When RNA extraction or flow sorting batches are visualized on the PCA graph, a batch effect is clearly present (see Supplementary Figures). As a result of batch effect, the expression profiles of non-infected do not separate cleanly from infected animals (**Figure 1a**).  

\quad With batch correction non-infected and infected animals separate cleanly (**Figure 1b**). Subsequent analyses use batch-corrected reads.

```{r echo = FALSE, fig.cap = "After batch correction, infected and non-infected animals cluster independently."}
experiment_mod <- model.matrix( ~ animals_merged$timepoint + animals_merged$infected + animals_merged$sive + animals_merged$aids + animals_merged$plus_minus) #full model

#normalizing reads, removing batch effect
dge_norm <- cpm(dge, log=TRUE, prior.count=3)
batch_corrected <- removeBatchEffect(dge_norm, batch=animals_merged$RNA_date, batch2 = animals_merged$sort_date, design = experiment_mod)
#calculating principal components and merging with metadata table
pca_corrected <- prcomp(t(batch_corrected))
pca_corrected_matrix <- as.data.frame(pca_corrected$x)
pca_corrected_matrix_merged <- merge(pca_corrected_matrix, animals_merged, by=0)

#non-batch corrected, for plotting purposes
raw_matrix <- pca_extract(dge)
#piping is not working and I don't want to troubleshoot this right now
raw_matrix_merged <- as.data.frame(raw_matrix$x) %>%
  merge(., animals_merged, by=0)


pca_plot <- function(data, group){
  group <- sym(group)
  
  ggplot(data, aes(PC1, PC2, color = !!group)) + 
  geom_point() +
  theme_bw()
}

plot_one_a <- pca_plot(raw_matrix_merged, "infected") +
  labs(title = "1a",
       caption = "Prior to batch correction")

plot_one_b <- pca_plot(pca_corrected_matrix_merged, "infected") +
  labs(title = "1b",
       caption = "After batch correction")

figure_one <- grid.arrange(plot_one_a, plot_one_b, ncol = 2, nrow = 1)
```

# The expression profile of BrdU+ monocytes change substantially with infection.

\quad Two graphs are presented. On the left, the PCA graph has been colored by day post infection; 0dpi corresponds to non-infected animals (**Figure 2a**). On the right, the PCA graph has been colored by BrdU+ vs. BrdU- monocytes (**Figure 2b**. The expression profile of BrdU+ monocytes change substantially with infection, while BrdU-negative monocytes are more similar between non-infected and infected animals. 

```{r echo = FALSE, fig.cap = "Changes in BrdU+ and BrdU- expression profiles with infection"}
plot_two_a <- pca_plot(pca_corrected_matrix_merged, "timepoint") +
  labs(title = "2a")
plot_two_b <- pca_plot(pca_corrected_matrix_merged, "plus_minus") +
  labs(title = "2b")
grid.arrange(plot_two_a, plot_two_b, ncol=2, nrow=1)

```
\newpage

# BrdU+ expression profiles at 21 days post-infection predict SIVE development
\quad SIV-infected animals with increased %BrdU+ monocytes at days 21 post infection rapidly progress to AIDS and develop severe encephalitis. The identity of BrdU+ monocytes is undefined. In this section, PCA graphs of BrdU+ monocytes from SIV-infected animals are presented.  

\quad The leftmost graph shows all BrdU+ monocytes from SIV-infected animals, annotated by day post infection (**Figure 3a**). The middle graph shows only 21dpi BrdU+ monocytes and is annotated by SIVE status(**Figure 3b**). The rightmost graph displays only necropsy timepoint BrdU+ monocytes and is annotated by SIVE status (**Figure 3c**).  

\quad BrdU+ monocytes at 21dpi and necropsy cluster separately (**Figure 3a**). At 21dpi, Animals that develop SIVE vs. SIVnoE cluster separately on the basis of BrdU+ monocyte expression profiles (**Figure 3b**). At necropsy, BrdU+ expression profiles are similar between SIVE and SIVnoE animals (**Figure 3c**). This suggests early monocyte dysregulation plays a key role in SIVE. This notion is consistent with Burdo (2011). The similarity of monocyte expression profiles at necropsy is consistent with Nowlin (2018).

```{r echo = FALSE, fig.cap = "BrdU+ monocyte expression profile at 21dpi predicts SIVE", fig.height = 9}
#dge_norm contains normalized read counts
#removing non-infected animals to look at differences among SIV+ animals better
#steps: 1) identify infected, brdu+ sample ids 2) selected infected animals 3) perform pca

#generate a list of samples that are brdu and infected

brdu_inf <- animals_merged %>%
  select(infected, plus_minus) %>%
  filter(infected == "Y" & plus_minus == "plus") %>%
  t()

#pretty nifty. subsets only infected, brduplus samples, then performs prcomp
pca_inf_plus <- batch_corrected %>%
  subset(TRUE, colnames(brdu_inf)) %>%
  t() %>%
  prcomp() %>%
  #pipe operator is my savior 
  (function(i) i$x)() %>%
  as.data.frame() %>%
  #merging with metadata
  merge(., y = animals_merged, by = 0) %>%
  #removing the duplicate sample (used for batch correction) for visualization
  filter(!Row.names == "sample_27") %>%
  arrange(timepoint)

brdu_early <- pca_inf_plus %>%
  filter(timepoint == "21dpi")

brdu_late <- pca_inf_plus %>%
  filter(timepoint == "nec")

plot_3_a <- pca_plot(pca_inf_plus, "timepoint") +
  labs(title = "3a",
       caption = "BrdU+ monocytes cluster by timepoint")

plot_3_b <- pca_plot(brdu_early, "sive") +
  labs(title = "3b",
       caption = "BrdU+ monocytes between SIVE vs. SIVnoE animals cluster separately at 21dpi")

plot_3_c <- pca_plot(brdu_late, "sive") +
  labs(title = "3c",
       caption = "BrdU+ monocytes are more similar at necropsy, and do not separate by SIVE status")

figure_three <- grid.arrange(plot_3_a, plot_3_b, plot_3_c, 
                             ncol=1, 
                             nrow=3)
```

#Differential expressions
To identify features of BrdU+ monocytes that predict rapid disease progression, differential expression was performed to compare BrdU+ monocytes from SIVE vs. SIVnoE animals at 21dpi.

```{r differential expression, echo = FALSE}

#using reads without batch correction - batches are included in model as recommended by voom creator

#BIG THING NEED TO FIX THIS. YOU HAVEN'T SUBSETTED FOR 21 DPI.
counts_brdu_21dpi <- count_data %>%
  select(c("Geneid","Gene_symbol",colnames(brdu_inf)))

dge_brdu_21dpi <- normalize_reads(counts_brdu_21dpi)

animals_brdu_21dpi <- animals %>%
  filter(rownames(.) %in% as.list(colnames(brdu_inf)))

#temporarily subsetting
dge_brdu_21dpi_sub <- dge_brdu_21dpi[1:1000, ]
test <- dge_brdu_21dpi[1:1000, ]

form <- ~ sive + RNA_date + (1|animal)
#what in the fuck is going on here, my columns in data match my sample rows. but it says it does not?
#is my metadata wrong?
#wait. does my metadata have column names
#oh. my metadata is empty
#

v_obj_dream <- voomWithDreamWeights(dge_brdu_21dpi_sub, form, animals_brdu_21dpi, BPPARAM = param)
# Specify parallel processing parameters
# this is used implicitly by dream() to run in parallel

param <- SnowParam("SOCK", progressbar = TRUE)

#Specify model 


#estimate weights using linear mixed model
fitmm <- dream(v_obj_dream, form, animals_brdu_21dpi)
fitmm <- eBayes(fitmm)

topTable(fitmm, coef = "siveY", number=Inf) %>%
  arrange(desc(logFC)) %>%
  arrange(adj.P.Val)

```

# Supplementary data
## Batch correction
\quad Samples extracted 2024-06-28 (cyan) and 2024-05-28 (red) cluster very far apart. These batches are balanced with respect to infection and therefore should cluster together. A similar effect is present for flow sorting batches.

```{r rna_batching, echo = FALSE, fig.cap = "batch effect from RNA extraction"}
#calculating and extracting matrix of principal components
pca_data <- pca_extract(dge)
pca_matrix <- as.data.frame(pca_data$x)

#merging principal components with animal metadata
pca_matrix_merged <- merge(pca_matrix, animals_merged, by=0)

sup_one <- ggplot(pca_matrix_merged, aes(PC1, PC2, color=RNA_date)) +
  geom_point() +
  theme_bw() +
  labs(title = "S1a",
       caption = "PCA graph annotated by RNA batch")

sup_two <- ggplot(pca_matrix_merged, aes(PC1, PC2, color=sort_date)) +
  geom_point() +
  theme_bw() + 
  labs(title = "S2a",
       caption = "PCA graph annotated by FACS batch")

sup_fig_one <- grid.arrange(sup_one, sup_two, ncol=2, nrow=1)

```


