---
title: "BrdU RNA-seq"
author: "Andrew Ding-Su"
date: "2024-11-08"
output: pdf_document
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)

#library calls
library("tidyverse")
library("variancePartition")
library("edgeR")
library("BiocParallel")
library("readxl")

source("./scripts/1 normalize.R")
source("./scripts/2 analysis.R")
source("./scripts/3 graphing.R")

```

```{r}
#Load data
count_data <- read.csv("./data/countMatrix.csv")
animals <- read_xlsx("./data/library_info.xlsx") 
brdu_values <- read.csv("./data/brdu_values.csv")

```
```{r cleaning, include = FALSE}
animals <- animals %>%
  mutate(sample = `Sample no.`, 
         animal = factor(Animal), 
         timepoint = factor(Timepoint, levels = c("0","21dpi","nec")),
         cd8_depleted = factor(cd8_depleted, levels = c("N","Y")),
         infected = factor(infected, levels = c("N","Y")),
         aids = factor(aids, levels = c("N","Y")),
         sive = factor(sive, levels = c("N","Y")),
         plus_minus = factor(`plus/minus`, levels = c("minus","plus")),
         library_date = factor(library_date),
         sort_date = factor(sort_date),
         RNA_date = factor(RNA_date),
         .keep = "none") %>%
  
  #next steps are to get the sample names to match
  mutate(sample = ifelse(sample <= 9, paste0("0",sample), sample)) %>%
  mutate(sample = paste0("sample_",sample))
animals <- as.data.frame(animals)
rownames(animals) <- animals$sample

#fixing data from brdu_values
brdu_values <- brdu_values %>%
  mutate(sample = `Sample.no.`) %>%
  mutate(sample = ifelse(sample <= 9, paste0("0",sample), sample)) %>%
  mutate(sample = paste0("sample_",sample))
rownames(brdu_values) <- brdu_values$sample

#Create full metadata sheet by merging animals with brdu_values
animals_merged <- merge(animals, brdu_values, by=0, all=TRUE) %>%
  mutate(sample = sample.x,
         animal,
         timepoint,
         cd8_depleted,
         infected,
         aids,
         sive,
         plus_minus,
         library_date,
         sort_date,
         RNA_date,
         per_brdu,
         .keep = "none")
rownames(animals_merged) <- animals_merged$sample

#generate dgelist object for analyses
dge <- normalize_reads(count_data)
```

##Principal component analyses

Clustering is first used for quality control. Then, clustering is used to visualize similarities and differences between samples. These similarities and differences are used to guide differential expression analyses.

#Quality control
After batch correction (see supplemental figures), non-infected and infected animals cluster separately as expected. Within non-infected animals, cd8-depleted and non-depleted animals cluster separately. This indicates that batch correction was performed properly and data quality is good. 

```{r PCAquality, echo = false}
experiment_mod <- model.matrix( ~ animals_merged$timepoint + animals_merged$infected + animals_merged$sive + animals_merged$aids + animals_merged$plus_minus) #full model

#normalizing reads, removing batch effect
dge_norm <- cpm(dge, log=TRUE, prior.count=3)
batch_corrected <- removeBatchEffect(dge_norm, batch=animals_merged$RNA_date, batch2 = animals_merged$sort_date, design = experiment_mod)
#calculating principal components and merging with metadata table
pca_corrected <- prcomp(t(batch_corrected))
pca_corrected_matrix <- as.data.frame(pca_corrected$x)
pca_corrected_matrix_merged <- merge(pca_corrected_matrix, animals_merged, by=0)

pca_plot <- function(data, group){
  group <- sym(group)
  
  ggplot(data, aes(PC1, PC2, color = !!group)) + 
  geom_point() +
  theme_bw()
}

pca_plot(pca_corrected_matrix_merged, "infected")
pca_plot(pca_corrected_matrix_merged, "cd8_depleted")
```
#The expression profile of BrdU+ monocytes change substantially with infection.
The expression profile of BrdU+ monocytes change substantially with infection. BrdU-negative monocytes are similar between non-infected and infected animals. 
```{r PCAdifferences, echo = false}
pca_plot(pca_corrected_matrix_merged, "timepoint")
pca_plot(pca_corrected_matrix_merged, "plus_minus")

```

The following analyses focus on BrdU+ monocytes. 
#BrdU+ monocytes move further away from normal with SIVE development, and return to a "normal" expression profile without SIVE. 
Arrows connect individuals from day 21 post infection to necropsy. Non-infected animals are located on the left of the PCA plot.

```{r echo = false}
pca_brduplus <- pca_corrected_matrix_merged %>%
  filter(plus_minus == "plus") %>%
  filter(!Row.names == "sample_27") %>%
  arrange(timepoint)

pca_plot(pca_brduplus, "sive") +
  geom_path(aes(x = PC1, y = PC2, group=animal),
            arrow = arrow(length = unit(0.55,"cm")))

```

#BrdU+ expression profiles at dpi21 predict SIVE development
BrdU+ monocytes cluster timepoint and by SIVE status
This graph is a PCA of only BrdU+ monocytes from infected animals. The purpose is to better visualize differences among infected animals.

```{r echo = false}
#dge_norm contains normalized read counts
#removing non-infected animals to look at differences among SIV+ animals better
#steps: 1) identify infected, brdu+ sample ids 2) selected infected animals 3) perform pca

#generate a list of samples that are brdu and infected

brdu_inf <- animals_merged %>%
  select(infected, plus_minus) %>%
  filter(infected == "Y" & plus_minus == "plus") %>%
  t()

#pretty nifty. subsets only infected, brduplus samples, then performs prcomp
pca_inf_plus <- batch_corrected %>%
  subset(TRUE, colnames(brdu_inf)) %>%
  t() %>%
  prcomp() %>%
  #pipe operator is my savior 
  (function(i) i$x)() %>%
  as.data.frame() %>%
  #merging with metadata
  merge(., y = animals_merged, by = 0) %>%
  #removing the duplicate sample (used for batch correction) for visualization
  filter(!Row.names == "sample_27") %>%
  arrange(timepoint)

pca_plot(pca_inf_plus, "per_brdu") +
  scale_color_gradient(low = "white", high="blue")

pca_plot(pca_inf_plus, "sive") +
  geom_path(aes(x = PC1, y = PC2, group=animal),
            arrow = arrow(length = unit(0.55,"cm")))

pca_plot(pca_inf_plus, "aids") +
  geom_path(aes(x = PC1, y = PC2, group=animal),
            arrow = arrow(length = unit(0.55,"cm")))

pca_plot(pca_inf_plus, "timepoint")+
 geom_path(aes(x = PC1, y = PC2, group=animal),
            arrow = arrow(length = unit(0.55,"cm")))

pca_plot(pca_inf_plus, "animal") +
  geom_path(aes(x = PC1, y = PC2, group=animal),
            arrow = arrow(length = unit(0.55,"cm")))
```
#BrdU+ monocytes from SIVnoE animals cluster separately from SIVnoE animals at 21dpi
We have previously shown that the percentage of BrdU+ monocytes early after infection predicts rapid disease progression and severe SIVE. Here, we find that BrdU+ monocytes from SIVE animals cluster distinctly from SIVnoE animals at 21dpi. This is consistent with an important role for early monocyte dysregulation in SIVE development.


## Notes to self
#TODO: repeat the above, but with BrdU- monocytes?
# Make comparisons in monocytes that predict SIVE, it looks like rapid progressors are different than slow progressors
#Split up the graphs to make it easier to see
# Looks like BrdU pos and negative are pretty similar at all timepoints, but they trend different over time, if that makes sense. So make timepoint and sive comparisons
```{r modeling}


# Specify parallel processing parameters
# this is used implicitly by dream() to run in parallel
param <- SnowParam(4, "SOCK", progressbar = TRUE)

#Specify model 
max_form <- ~ infected + cd8_depleted + aids + sive + plus_minus + (1|animal)
interact_form <- ~ cd8_depleted + infected + timepoint*plus_minus + (1|animal)

#estimate weights using linear mixed model
v_obj_dream <- voomWithDreamWeights(dge, form, animals, BPPARAM = param)
#fitmm <- dream(v_obj_dream, form, animals)
#fitmm <- eBayes(fitmm)

topTable(fitmm, coef = "plus_minusplus", number=Inf)

v_obj_interact <- voomWithDreamWeights(dge, interact_form, animals, BPPARAM = param)
fit_interact <- dream(v_obj_interact, interact_form, animals)
fit_interact <- eBayes(fit_interact)

topTable(fit_interact, coef = "timepointnec:plus_minusplus", number=Inf) %>%
  filter(P.Value <= 0.05) %>%
  arrange(desc(logFC))
```

## Supplementary data
# Batch correction
Samples extracted 2024-06-28 (cyan) and 2024-05-28 (red) cluster very far apart. These two batches had animals with virtually the same conditions and therefore should cluster together. Therefore, I batch corrected by RNA extraction date

```{r rna_batching, echo = FALSE, fig.cap = "batch effect from RNA extraction"}
#calculating and extracting matrix of principal components
pca_data <- pca_extract(dge)
pca_matrix <- as.data.frame(pca_data$x)

#merging principal components with animal metadata
pca_matrix_merged <- merge(pca_matrix, animals_merged, by=0)

ggplot(pca_matrix_merged, aes(PC1, PC2, color=RNA_date)) +
  geom_point() +
  theme_bw()

```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
